parlai/chat_service/utils/timeout.py

```diff
@@ -53,6 +53,7 @@ def _is_valid(act):
                         "quick_replies": quick_replies,
                     }
                 )
+            time.sleep(1)
```

Comment:
This is a fairly unresponsive sleep - definitely need to have one but `0.1` may be better for feeling snappy in response to user input (especially as this is shared between all kinds of chat services). Could also put it in a constant

Reply:
fwiw @JackUrb I also though 1 was too long but anecdotally I haven't felt the response time to be too long yet.

Reply:
Fair enough - ultimately when it comes to using polling whatever feels like it's not a bad tradeoff between responsiveness and interruption is alright. We've solved this in Mephisto using a `threading.Event()` that is set whenever an action is available, this way we can wait on that even rather than polling.

Reply:
1 was purely based off anecdotal evidence - indeed using a `threading.Event()` is the appropriate course of action here, but this was meant as a quick fix. I can reduce to `0.1` for now